### PR TITLE
[E2E refactor - 6] refactor collector e2e test

### PR DESCRIPTION
**Pull request by pintohutch**
I decided to break out the kind E2E test refactor PR https://github.com/GoogleCloudPlatform/prometheus-engine/pull/738 into smaller, digestible PRs for reviewing.

Note: because of the nature of the change, presubmits (i.e. Github Actions) may fail until the final PR is merged.

This is the sixth one, where we refactor and simplify collector e2e tests.

I've also added the ability to TLSInsecureScrapeVerify for kubelet scraping (though this will soon be legacy anyways).

### CODE REVIEW

**Review by macxamin — APPROVED** [review]
Amazing work 💪 !

File: pkg/operator/apis/monitoring/v1/types.go [hunk 1]
```diff
@@ -144,6 +144,9 @@ const CompressionGzip CompressionType = "gzip"
 type KubeletScraping struct {
 	// The interval at which the metric endpoints are scraped.
 	Interval string `json:"interval"`
+	// TLSInsecureSkipVerify disables verifying the target cert.
+	// This can be useful for clusters provisioned with kubeadm.
+	TLSInsecureSkipVerify bool `json:"tlsInsecureSkipVerify,omitempty"`
```
Comment (bwplotka) [inline pkg/operator/apis/monitoring/v1/types.go:149]: is this part of refactor?

File: e2e/collector_test.go [hunk 2]
```diff
@@ -24,405 +24,495 @@ import (
...
 	// more fine-grained stages makes debugging a lot easier.
-	t.Run("deployed", tctx.subtest(testCollectorDeployed))
-
-	t.Run("scrape", tctx.subtest(func(ctx context.Context, t *OperatorContext) {
-		t.createOperatorConfigFrom(ctx, monitoringv1.OperatorConfig{
-			Features: monitoringv1.OperatorFeatures{
-				TargetStatus: monitoringv1.TargetStatusSpec{
-					Enabled: true,
+	t.Run("collector-deployed", testCollectorDeployed(ctx, t, kubeClient))
+	t.Run("collector-configured", testCollectorConfigured(ctx, t, kubeClient, opClient))
```
Comment (bwplotka) [inline e2e/collector_test.go:52]: Same problem as in https://github.com/GoogleCloudPlatform/prometheus-engine/pull/747#discussion_r1474426870, maybe merge with `-configuration` job?

File: e2e/collector_test.go [hunk 3]
```diff
@@ -24,405 +24,495 @@ import (
...
-					Selector: metav1.LabelSelector{
-						MatchLabels: deployment.Spec.Template.Labels,
-					},
-					Endpoints: []monitoringv1.ScrapeEndpoint{
-						{
-							Port:     intstr.FromString(operatorutil.SyntheticAppPortName),
-							Interval: "5s",
-						},
-					},
+	// We could simply verify that the full collection chain works once. But validating
```
Comment (bwplotka) [inline e2e/collector_test.go:91]:
```suggestion
	// TODO(pintohutch): We could simply verify that the full collection chain works once. But validating
```

File: pkg/operator/apis/monitoring/v1/types.go (hunk 1 above)
Comment (pintohutch) [reply pkg/operator/apis/monitoring/v1/types.go:149]: Yea - I added that feature to kubelet scraping here. It's non-breaking and probably pretty fringe, but it helps us test in e2e for kubelet scraping.

I'll mark it out as a separate commit.

File: e2e/collector_test.go (hunk 2 above)
Comment (pintohutch) [reply e2e/collector_test.go:52]: Thanks! Renamed to emphasize we're updating the operatorconfig.

**Pull request merged**